native/arrow_format_nif/src/lib.rs

```diff
@@ -49,10 +46,156 @@ mod atoms {
     }
 }
 
+#[cfg(test)]
+pub mod test {
```

Comment:
I believe you could have this extracted to a `test.rs` file in the `src` directory. This is a non-blocker, though :)